ArjunaCore/arjuna/tests/byteman-scripts/recoverySuspend.btm

```diff
@@ -0,0 +1,56 @@
+########################################################################
+#
+# Byteman script used to make BytemanControlledRecord fail
+# topLevelCommit until AtomicActionRecoveryModule acknowledges that
+# it needs to block the suspension of PeriodicRecovery.
+# The end goal of this script is to simulate a transaction that fails
+# while PeriodicRecovery is trying to recover it. This scenario makes
+# sure that PeriodicRecovery waits to suspend until all transactions
+# have recovered
+#
+########################################################################
+
+########################################################################
+# Rule to clear greenFlag when invoking 
+# BytemanControlledRecord.resetGreenFlag()
+#########################################################################
+
+RULE resetFlag
+CLASS com.hp.mwtests.ts.arjuna.resources.BytemanControlledRecord
+METHOD resetGreenFlag
+AT ENTRY
+IF TRUE
+DO debug("[BytemanControlledRecord] clearing greenFlag"),
+   clear("greenFlag")
+ENDRULE
+
+########################################################################
+# Rule to create greenFlag when
+# AtomicActionRecoveryModule.shouldBlockShutdown returns true
+#########################################################################
+
+RULE setFlag
+CLASS com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule
+METHOD shouldBlockShutdown
+AT EXIT
+BIND returnValue:boolean = $!
+IF returnValue
+DO debug("[AtomicActionRecoveryModule] Setting greenFlag to true"),
+   flag("greenFlag")
```

Comment:
Can we only green flag if we know isWaitForFinalRecovery has been called in PeriodicRecovery::suspendScan?

Reply:
Super - thanks! (BTW I am not spotting a resolve thread type of ability but this thread is resolved :) )

Reply:
Thanks. Solving